src/Either.js

```diff
@@ -70,6 +70,23 @@ _Right.prototype.chain = function(f) {
   return f(this.value);
 };
 
+//chainRec
+Either.chainRec = Either.prototype.chainRec = function(f, i) {
+  var state = util.chainRecNext(i);
+  while (state.done === false) {
+    state = Either.either(
+      function(v) {
+        return { done: true, value: v, isLeft: true };
+      },
+      function(v) {
+        return v;
+      },
+      f(util.chainRecNext, util.chainRecDone, state.value)
```

Comment:
We could make use of the existing `Either` to handle `next` and `done` here.

I think something like this should work:
```js
Either.chainRec = Either.prototype.chainRec = function(f, i) {
  var updateState = Either.either(
    compose(Right, Left),
    Either.either(compose(Left, Right), compose(Right, Right))
  );
  var state = Left(Right(i));
  while (Either.isLeft(state)) {
    state = updateState(f(Left, Right, state.value.value));
  }
  return state.value;
};
```